package/src/lib.rs

```diff
@@ -0,0 +1,27 @@
+//! The [Package] containing a [vm_core::Program] or [assembly::Library] and a manifest(exports and
+//! dependencies).
+
+#![no_std]
+
+extern crate alloc;
+
+mod dep;
+mod package;
+
+#[cfg(test)]
+extern crate std;
+#[cfg(test)]
+mod tests;
+
+pub use self::{
+    dep::{
+        resolver::{
+            DependencyResolver, LocalResolvedDependency, MemDependencyResolverByDigest,
+            ResolvedDependency,
+        },
+        Dependency, DependencyName,
+    },
+    package::{MastArtifact, Package, PackageExport, PackageManifest},
+};
```

Comment:
Question: should we re-export such things as `Digest`, `Program`, `MastForest` etc. from here is that this crate is more "self-contained"?

Reply:
I'd say "yes". I can imagine `miden-package` crate being used for some sort of package analysis without relaying on the rest of VM crates.

Reply:
Done.